document-store/src/main/java/org/hypertrace/core/documentstore/Collection.java

```diff
@@ -102,6 +104,14 @@
    */
   CreateResult create(Key key, Document document) throws IOException;
 
+  /**
+   * Updates existing documents if the corresponding Filter condition evaluates to true
+   * @param documents to be updated in bulk
+   * @return true if operation succeeded
+   */
+  boolean bulkUpdate(List<Triple<Key, Document, Filter>> documents);
```

Comment:
Here, we don't need any conditional updates, right?
And does `boolean bulkUpsert(Map<Key, Document> documents);` suffice? or we don't want to insert if the document is not present?

Reply:
I need conditional updates (update a timestamp field only if it is greater than what is already there in DB) and I don't want to insert if a record is not there in DB